### PR TITLE
organisations: show organisation information

### LINKED ISSUE #268: Organisation landing page
* Create a landing page for organisation, showing information about it (name, logo, description).
* Add logo and description to the editor for organisation.

### PR DESCRIPTION
**Pull request by sebastiendeleze**
This commit shows information about the current organisation, as the title, the description and optionally a logo. Users can search for records that belong to this organisation only.

* Enables REST endpoints for organisations files.
* Enables UI endpoints for organisations files.
* Associates un bucket when an organisation is created.
* Adds `description`, `_bucket` and `_files` properties to JSON schema.
* Adds `description` property to elasticsearch mapping.
* Serializes `description`, `_bucket` and `_files` properties when organisation is retrieved by REST API.
* Adds a Jinja filter to get an image URL for a record.
* Displays organisation's information in front page and inner pages.
* Searches only for global records in main search input.
* Closes #268.

Co-Authored-by: Sébastien Délèze <sebastien.deleze@rero.ch>

### CODE REVIEW

File: sonar/theme/templates/sonar/partial/navbar.html [hunk 1]
```diff
@@ -58,7 +43,7 @@
           class="d-inline-block align-top mr-3 my-4" alt="">
       </a>
       {% if not admin %}
-      <form action="{{ url_for('documents.search', view=view_code) }}"
+      <form action="{{ url_for('documents.search', view=view_code if g.get('organisation', {}).get('isDedicated') else config.SONAR_APP_DEFAULT_ORGANISATION) }}"
```
Comment (zannkukai) [inline sonar/theme/templates/sonar/partial/navbar.html:46]: you should add a default return to `False` for the `get('isDedicated')`. If the `get('organisation', {})` return an empty dict, then `get('isDedicated')` will failed.

File: sonar/theme/templates/sonar/frontpage.html [hunk 2]
```diff
@@ -24,7 +24,7 @@
     <div class="container">
       <div class="row justify-content-center">
         <div class="col-8 col-xs-6 col-lg-4 py-3 py-sm-5">
-            <a href="{{ url_for('documents.index', view=view_code) }}">
+            <a href="{{ url_for('documents.index', view=view_code if g.get('organisation', {}).get('isDedicated') else config.SONAR_APP_DEFAULT_ORGANISATION) }}">
```
Comment (zannkukai) [inline sonar/theme/templates/sonar/frontpage.html:27]: same here

File: sonar/theme/templates/sonar/frontpage.html [hunk 3]
```diff
@@ -36,7 +36,7 @@
       </div>
       <div class="row justify-content-center">
           <div class="col-sm-12 col-lg-8 text-right my-4">
-            <form class="justify-content-end" action="{{ url_for('documents.search', view=view_code) }}" role="search">
+            <form class="justify-content-end" action="{{ url_for('documents.search', view=view_code if g.get('organisation', {}).get('isDedicated') else config.SONAR_APP_DEFAULT_ORGANISATION) }}" role="search">
```
Comment (zannkukai) [inline sonar/theme/templates/sonar/frontpage.html:39]: here too

File: sonar/theme/views.py [hunk 4]
```diff
@@ -189,6 +189,39 @@ def schemas(record_type):
...
+                                                  bucket=file['bucket'])
+
+    for file in record['_files']:
+        if re.match(r'^.*\.(jpe?g|png|gif|svg)$',
+                    file['key'],
+                    flags=re.IGNORECASE):
+            if not key:
+                return image_url(file)
+
+            if file['key'] == key:
```
Comment (zannkukai) [inline sonar/theme/views.py:219]: COSMETIC : maybe merge both if statement.

File: sonar/theme/templates/sonar/partial/navbar.html (hunk 1 above)
Comment (sebastiendeleze) [reply sonar/theme/templates/sonar/partial/navbar.html:46]: I think it's not correct, this instruction will never fail. If the organisation is an empty object, `isDedicated` is resolved as `None` which is the same behavior than `False` in this case. What do you think about it?

Here's a quick example: https://repl.it/@sebastiendeleze/test

Comment (zannkukai) [reply sonar/theme/templates/sonar/partial/navbar.html:46]: I'm agree ; it's just not really human readable.

**Review by zannkukai — APPROVED** [review]

**Comment by mmo** [comment]
- suggestion: add info "HTML markup admitted" next to the "Description" text area
- problem: HTML markdown artificially adds <br> tags

**Comment by mmo** [comment]
problem: not possible to add an SVG image

**Pull request merged**